back/engines/commercial/id_vienna_saml/app/lib/id_vienna_saml/citizen_saml_omniauth.rb

```diff
@@ -72,5 +76,32 @@ def redirect_uri(configuration)
     def vienna_login_env
       AppConfiguration.instance.settings('vienna_citizen_login', 'environment').to_sym
     end
+
+    # Generates a placeholder name based on the email.
+    # For first names, uses the first character of the email,
+    # For last names, uses the second character of the email or the first character of the second word if available.
+    # @param [String] email
+    # @param [Symbol] part Either `:first` or `:last`
+    # @return [String]
+    def generate_placeholder_from_email(email, part)
+      case part
+      when :first
+        placeholder = email[0]
+      when :last
+        local_part = email.split('@').first
+        words = local_part.split(/_|\./)
+        if words.size > 1
+          second_word = words[1]
+          placeholder = second_word[0]
```

Comment:
```suggestion
          placeholder = second_word
```
I don't think second word can ever have more than one part.

Reply:
This is the first character of the string.

Reply:
Ah yes, true! I forgot that!